service/history/task/cross_cluster_target_task_executor.go

```diff
@@ -0,0 +1,321 @@
+)
+
+type (
+	crossClusterTargetTaskExecutor struct {
+		shard                   shard.Context
+		logger                  log.Logger
+		metricsClient           metrics.Client
+		historyClient           history.Client
+		parentClosePolicyClient parentclosepolicy.Client
+		config                  *config.Config
+	}
+)
+
+// NewCrossClusterTargetTaskExecutor creates a new task Executor for
+// processing cross cluster tasks at target cluster
+func NewCrossClusterTargetTaskExecutor(
+	shard shard.Context,
+	logger log.Logger,
+	config *config.Config,
+) Executor {
+	return &crossClusterTargetTaskExecutor{
+		shard:         shard,
+		logger:        logger,
+		metricsClient: shard.GetMetricsClient(),
+		historyClient: shard.GetService().GetHistoryClient(),
+		parentClosePolicyClient: parentclosepolicy.NewClient(
+			shard.GetMetricsClient(),
+			shard.GetLogger(),
+			shard.GetService().GetSDKClient(),
+			config.NumParentClosePolicySystemWorkflows(),
+		),
+		config: config,
+	}
+}
+
+// Execute executes a cross cluster task at the target cluster.
+//
+// task.response field will be populated as long as the passed in
+// task has the right type (*crossClusterTargetTask).
+//
+// Returned error will be non-nil only when the error is retryable.
+// Non-retryable errors, such as domain/workflow not exists or
+// errMissingTaskRequestAttributes, will not be returned,
+// as when those error happens the task execution is considered as
+// completed, and the error will be returned to the source cluster
+// via the failedCause field in task.response
+func (t *crossClusterTargetTaskExecutor) Execute(
+	task Task,
+	shouldProcessTask bool,
+) error {
+	targetTask, ok := task.(*crossClusterTargetTask)
+	if !ok {
+		return errUnexpectedTask
+	}
+
+	response := &types.CrossClusterTaskResponse{
+		TaskID:   targetTask.GetTaskID(),
+		TaskType: targetTask.request.TaskInfo.TaskType,
+	}
+
+	if !shouldProcessTask {
+		// this should not happen,
+		// shouldProcessTask parameter will be deprecated after
+		// 3+DC task lifecycle is done.
+		response.FailedCause = types.CrossClusterTaskFailedCauseUncategorized.Ptr()
+		targetTask.response = response
+		return errors.New("encounter shouldProcessTask equals to false when processing cross cluster task at target cluster")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), taskDefaultTimeout)
+	defer cancel()
+
+	var err error
+	switch task.GetTaskType() {
+	case persistence.CrossClusterTaskTypeStartChildExecution:
+		response.StartChildExecutionAttributes, err = t.executeStartChildExecutionTask(ctx, targetTask)
+	case persistence.CrossClusterTaskTypeCancelExecution:
+		response.CancelExecutionAttributes, err = t.executeCancelExecutionTask(ctx, targetTask)
+	case persistence.CrossClusterTaskTypeSignalExecution:
+		response.SignalExecutionAttributes, err = t.executeSignalExecutionTask(ctx, targetTask)
+	default:
+		err = errUnknownTransferTask
```

Comment:
Can we use our own errors? `errUnknownCrossClusterTask`

Reply:
Updated.